File: Excel/441 Challenge.py
import pandas as pd
import re

input = pd.read_excel("441 Integer Intervals.xlsx", usecols="A", nrows=7)
test = pd.read_excel("441 Integer Intervals.xlsx", usecols="B", nrows=7)

result = input.copy()
result['Problem'] = result['Problem'].str.split(", ")
result['row_number'] = result.index
result = result.explode('Problem')
result['Problem'] = result['Problem'].apply(lambda x: list(range(int(x.split('-')[0]), int(x.split('-')[1])+1)) if '-' in x else [int(x)])
result = result.explode('Problem')
result = result.groupby(result.index).agg({'Problem': lambda x: ', '.join(map(str, sorted(set(x))))}).reset_index(drop=True)

print(result['Problem'].equals(test['Answer Expected'])) # True

File: Excel/441 Challenge.R
library(tidyverse)
library(readxl)

input = read_excel("Excel/441 Integer Intervals.xlsx", range = "A1:A7")
test  = read_excel("Excel/441 Integer Intervals.xlsx", range = "B1:B7")

result = input %>%
  mutate(rn = row_number()) %>%
  separate_rows(Problem, sep = ", ") %>%
  mutate(Problem = map(Problem, ~{
    if(str_detect(., "-")){
      range = str_split(., "-")[[1]]
      seq(as.numeric(range[1]), as.numeric(range[2]))
    } else {
      as.numeric(.)
    }
  })) %>%
  unnest(Problem) %>%
  summarise(`Answer Expected` = str_c(sort(unique(Problem)), collapse = ", "), .by = rn) %>%
  select(-rn)

identical(result, test)
# [1] TRUE
